src/core/grid_based_algorithms/lbgpu_cuda.cu

```diff
@@ -216,7 +218,26 @@ __device__ void gaussian_random(LB_randomnr_gpu *rn) {
   rn->randomnr[0] = x2 * fac;
   rn->randomnr[1] = x1 * fac;
 }
-/* wrapper */
+
+static unsigned int philox_counter = 0;
+__device__ float4 random_wrapper_philox(unsigned int index, unsigned int mode,
+                                        unsigned int philox_counter) {
+  unsigned int xyz[3];
+  index_to_xyz(index, xyz);
```

Comment:
Just use the index directly and set the other two parameters to zero. This just wastes expensive division operations.